### PR TITLE
Perform unit tests on travis

### PR DESCRIPTION
**Pull request by avrabe**
Added a first unit test case for the tl_detector.
Added the testcase runner for ros as part of the CI.
Fixed a dependency topic on tl_detector:
 - Provide the publisher before the subscriber
   One a /image_color message is received, it is
   used to publish a traffic_waypoint

Signed-off-by: Ralf Anton Beier <ralf_beier@me.com>

### CODE REVIEW

**Comment by vguerra** [comment]
lgtm. 

+1 for start adding tests :)

File: ros/src/tl_detector/CMakeLists.txt [hunk 1]
```diff
@@ -200,4 +200,8 @@ include_directories(
 # endif()
 
 ## Add folders to be run by python nosetests
-# catkin_add_nosetests(test)
+#catkin_add_nosetests(test)
+if(CATKIN_ENABLE_TESTING)
+  find_package(rostest REQUIRED)
+  add_rostest(test/test_tl_detector.launch)
+endif()
```
Comment (asimonov) [inline ros/src/tl_detector/CMakeLists.txt:207]: why would github show this 'no newline at the end of the file' warning? could it affect anything in any language?

**Review by asimonov — APPROVED** [review]
great work!

File: ros/src/tl_detector/CMakeLists.txt (hunk 1 above)
Comment (avrabe) [reply ros/src/tl_detector/CMakeLists.txt:207]: I saw in the past some tools having some problems with it. According to the [stackoverflow](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) discussion, there are actually some implications with it.

**Pull request merged**